File: src/main/java/com/dkp/exchange/model/Coin.java
package com.dkp.exchange.model;

import jakarta.persistence.*;
import lombok.Data;
import java.math.BigDecimal;

@Data
@Entity
@Table(name = "coins")
public class Coin {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false)
    private String symbol;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private BigDecimal currentPrice;

    @Column(nullable = false)
    private BigDecimal priceChange24h;

    @Column(nullable = false)
    private BigDecimal volume24h;

    @Column(nullable = false)
    private BigDecimal marketCap;

    private boolean isActive = true;
} 

File: frontend/src/types/coin.ts
export interface PriceUpdate {
    symbol: string;
    price: number;
    volume24h: number;
    priceChange24h: number;
    timestamp: string;
}

export interface Coin {
    id: number;
    symbol: string;
    name: string;
    currentPrice: number;
    volume24h: number;
    marketCap: number;
    isActive: boolean;
} 